gossipd/routing.c

```diff
@@ -1064,20 +1064,30 @@ bool routing_add_channel_update(struct routing_state *rstate,
 	u32 timestamp;
 	u8 message_flags, channel_flags;
 	u16 expiry;
-	u64 htlc_minimum_msat;
+	u64 htlc_minimum_msat = 0; // init as 0
 	u32 fee_base_msat;
 	u32 fee_proportional_millionths;
 	u64 htlc_maximum_msat;
 	struct bitcoin_blkid chain_hash;
 	struct chan *chan;
 	u8 direction;
 
-	if (!fromwire_channel_update_option_channel_htlc_max(update, &signature, &chain_hash,
+	if (!fromwire_channel_update(update, &signature, &chain_hash,
 				     &short_channel_id, &timestamp,
 				     &message_flags, &channel_flags,
 				     &expiry, &htlc_minimum_msat, &fee_base_msat,
-				     &fee_proportional_millionths,
-				     &htlc_maximum_msat))
+				     &fee_proportional_millionths))
+		return false;
+	/* If it's flagged as containing the optional field, reparse for
+	 * the optional field */
+	if (message_flags & ROUTING_OPT_HTLC_MAX_MSAT &&
```

Comment:
is there a nicer way to do this, other than re-parsing once we find out the flag is set for it?